src/main/java/de/qabel/desktop/repository/persistence/PersistenceContactRepository.java

```diff
@@ -57,6 +57,21 @@ public void save(Contact contact, Identity identity) throws PersistenceException
 	}
 
 	@Override
+	public void delete(Contact contact, Identity identity) throws PersistenceException {
+		try {
+			Contacts personalContacts = contacts.get(identity.getKeyIdentifier());
+			personalContacts.remove(contact);
+
+			if (!persistence.updateOrPersistEntity(personalContacts)) {
+				throw new PersistenceException("Failed to delete Entity " + contact + ", reason unknown");
```

Comment:
fyi:
this exception is catched in line 69. The result would be:

Failed to delete Entity: Failed to delete Entity ;)